ACE/ace/SSL/SSL_Context.cpp

```diff
@@ -274,39 +254,8 @@ ACE_SSL_Context::set_mode (int mode)
     case ACE_SSL_Context::SSLv23:
       method = ::SSLv23_method ();
       break;
-    case ACE_SSL_Context::TLSv1_client:
-      method = ::TLSv1_client_method ();
-      break;
-    case ACE_SSL_Context::TLSv1_server:
-      method = ::TLSv1_server_method ();
-      break;
-    case ACE_SSL_Context::TLSv1:
-      method = ::TLSv1_method ();
-      break;
-#if defined(TLS1_1_VERSION) && (TLS_MAX_VERSION >= TLS1_1_VERSION)
-    case ACE_SSL_Context::TLSv1_1_client:
-      method = ::TLSv1_1_client_method ();
-      break;
-    case ACE_SSL_Context::TLSv1_1_server:
-      method = ::TLSv1_1_server_method ();
-      break;
-    case ACE_SSL_Context::TLSv1_1:
-      method = ::TLSv1_1_method ();
-      break;
-#endif
-#if defined(TLS1_2_VERSION) && (TLS_MAX_VERSION >= TLS1_2_VERSION)
-    case ACE_SSL_Context::TLSv1_2_client:
-      method = ::TLSv1_2_client_method ();
-      break;
-    case ACE_SSL_Context::TLSv1_2_server:
-      method = ::TLSv1_2_server_method ();
-      break;
-    case ACE_SSL_Context::TLSv1_2:
-      method = ::TLSv1_2_method ();
-      break;
-#endif
     default:
-      method = ::SSLv3_method ();
+      method = ::SSLv23_method ();
```

Comment:
On Thu, Dec 17, 2015 at 03:28:37PM -0800, obruns wrote:
> Exactly that is my key concern: I don't want to talk to a client (or
> server for that matter) that tries to force insecure protocols on me.
> But isn't it all there in stable OpenSSL already? I was doing this:
> 
> ```cpp
> ACE_SSL_Context tlsContext;
> /*
>  * Calls SSL_CTX_set_options() with anything but SSL_OP_NO_TLSv1_2
>  */
> int ret = tlsContext.filter_versions("tlsv1.2"); // TLSv1.2 only
> ```

I have no idea what that does, but I assume it sets things like
SSL_OP_NO_SSLv3.

> Unfortunately, the way
> `ACE_SSL_Context::filter_versions()` is implemented right now, it
> isn't forward-compatible, either.
> So this is what the new API, you referred to above, is for?

It's now available in the master branch:
https://www.openssl.org/docs/manmaster/ssl/SSL_CTX_set_min_proto_version.html

All relevant documentation should have been updated.